### PR TITLE
Improve README.md and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ebkalderon**
### Added

* Add simple instructions for how to install the `cargo2nix` CLI tool with `nix-env`.
* Document how to report build failures of crates from Crates.io in `CONTRIBUTING.md`.

### Changed

* Improve wording and formatting of `README.md` and `CONTRIBUTING.md`.
* Satisfy most lints from [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli).

### Removed

* Remove "airplane mode" section in `README.md` for the time being.

CC @port3000, since we were previously discussing how to install `cargo2nix` through `nix-env`.

### CODE REVIEW

**Comment by ebkalderon** [comment]
I'd recommend moving through the PR commit-by-commit for the best experience, and maybe view the rendered Markdown at the very end to see how the final results look. 😄

File: README.md [hunk 1]
```diff
@@ -122,21 +114,21 @@ for the `crates-io` public crates.
...
 ## Design
 
 This Nixpkgs overlay builds your Rust crates and binaries by first pulling the
 dependencies apart, building them individually as separate Nix derivations and
-linking them together. This is achieved by basically passing linker flags to the
-`cargo` invocations and the underlying `rustc`/`rustdoc` invocations.
+linking them together. This is achieved by passing custom linker flags to the
+`cargo` invocations and the underlying `rustc` and `rustdoc` invocations.
 
-The overlay is built on top of the work by James Kay. In addition, this overlay
```
Comment (eupn) [inline README.md]: Why the James Kay mention is removed?

Comment (ebkalderon) [reply README.md]: As mentioned in the commit message for c2c03f1, the same credit is repeated in the _Credits_ section immediately below.

**Comment by ebkalderon** [comment]
Okay, I think these improvements are safe enough to merge. Hopefully, I can make a few more user experience improvements to the documentation and to the repo itself in subsequent PRs.

**Pull request merged**